cookbooks/cdo-repository/recipes/default.rb

```diff
@@ -45,6 +47,12 @@
 
   user node[:user]
   group node[:user]
+
+  # Ensure proper environment for credential helper
+  environment(
+    'HOME' => home_path,
+    'USER' => node[:user]
+  )
```

Comment:
Did you confirm that this is necessary, or is this leftovers from experimentation? [The documentation](https://docs.chef.io/resources/git/) indicates that the default git provider will set `HOME` automatically, and I don't see anything in [our customizations](https://github.com/code-dot-org/code-dot-org/blob/staging/cookbooks/cdo-repository/libraries/git.rb) that I expect would break that